include/zephyr/smf.h

```diff
@@ -55,25 +38,15 @@
 	.parent  = _parent, \
 	.initial = _initial \
 }
-#endif /* CONFIG_SMF_INITIAL_TRANSITION */
-
 #else
-
-/**
- * @brief Macro to create a flat state.
- *
- * @param _entry  State entry function
- * @param _run  State run function
- * @param _exit  State exit function
- */
-#define SMF_CREATE_STATE(_entry, _run, _exit) \
+#define SMF_CREATE_STATE(_entry, _run, _exit, _parent, _initial) \
```

Comment:
This change should be squashed with the changes to the SMF_CREATE_STATE users.  Otherwise this breaks bisect-ability.